app/models/MessageResponse.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.XSDFile.GoodsReleasedXSD
+import models.XSDFile.UnloadingPermissionXSD
+
+sealed trait MessageResponse {
+  val messageReceived: MessageReceived
+  val messageType: MessageType
+  val xsdFile: XSDFile
+}
+
+case object GoodsReleasedResponse extends MessageResponse {
+  override val messageReceived: MessageReceived = MessageReceived.GoodsReleased
+  override val messageType: MessageType         = MessageType.GoodsReleased
+  override val xsdFile: XSDFile                 = GoodsReleasedXSD
+}
+case object UnloadingPermissionResponse extends MessageResponse {
+  override val messageReceived: MessageReceived = MessageReceived.UnloadingPermission
+  override val messageType: MessageType         = MessageType.UnloadingPermission
+  override val xsdFile: XSDFile                 = UnloadingPermissionXSD
+}
```

Comment:
I like that we've taken to model these as objects as opposed to strings floating around the code base 👍 the one downside is that we now seem to have `GoodsReleasedResponse`,  `MessageType.GoodsReleased`, `GoodsReleasedXSD` and `MessageReceived.GoodsReleased`. This here wants to unify some of them, but is grouping it together as a product type with some unrepresentable states, such as a 

 ```
MessageResponse {
  val messageReceived: MessageReceived = MessageReceived.GoodsReleased
  val messageType: MessageType         = MessageType.UnloadingPermission
  val xsdFile: XSDFile                 = ArrivalNotificationRejectionXSD
}
```

I think its a good first solution for this problem,  but we may need to refine it